tracer/test/Datadog.Trace.TestHelpers/MockTelemetryAgent.cs

```diff
@@ -129,25 +130,29 @@ public void Dispose()
             _listener?.Close();
         }
 
-        internal static TelemetryData DeserializeResponse(Stream inputStream, string apiVersion, string requestType)
+        internal static TelemetryData DeserializeResponse(Stream inputStream, string apiVersion, string requestType, bool compressed)
         {
             return apiVersion switch
             {
-                TelemetryConstants.ApiVersionV2 => DeserializeV2(inputStream, requestType),
+                TelemetryConstants.ApiVersionV2 => DeserializeV2(inputStream, requestType, compressed),
                 _ => throw new Exception($"Unknown telemetry api version: {apiVersion}"),
             };
 
-            static TelemetryData DeserializeV2(Stream inputStream, string requestType)
+            static TelemetryData DeserializeV2(Stream inputStream, string requestType, bool compressed)
             {
                 if (!TelemetryConverter.V2Serializers.TryGetValue(requestType, out var serializer))
                 {
                     throw new Exception($"Unknown V2 telemetry request type {requestType}");
                 }
 
-                using var sr = new StreamReader(inputStream);
-                var text = sr.ReadToEnd();
-                var tr = new StringReader(text);
```

Comment:
I wonder why we did this before, instead of what you did in this PR? 🤔 Seems odd that we would want to do that 😄 Maybe it was for easier debugging 🤷‍♂️